### PR TITLE
add local tests and static analysis workflows using GitHub Actions

### PR DESCRIPTION
**Pull request by mattsb42-aws**
*Description of changes:*

Initial conversion to GitHub Actions.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Comment by mattsb42-aws** [comment]
...aaand the workflows aren't triggering... :/

**Comment by mattsb42-aws** [comment]
Per offline research, I discovered that GA workflows do NOT trigger on the PR that creates the workflow file. These workflows will not trigger until a subsequent PR.

**Comment by scottarc** [comment]
Is that an issue with Github Actions itself, or how we're using it?

**Comment by mattsb42-aws** [comment]
> Is that an issue with Github Actions itself, or how we're using it?

Based on my testing, this is a fundamental characteristic of GitHub Actions. Presumably as a protection against random new workflows being added to repos without the maintainers' knowledge or permission.

**Comment by robin-aws** [comment]
Should the workflow be to create a fork from a repo and get the existing Actions, plus whatever new or updated ones you're contributing, to work on your fork, so that you can safely demonstrate the changes? How likely is that to be feasible?

**Comment by mattsb42-aws** [comment]
> Should the workflow be to create a fork from a repo and get the existing Actions, plus whatever new or updated ones you're contributing, to work on your fork, so that you can safely demonstrate the changes? How likely is that to be feasible?

That's the hard thing. I haven't figured what the best "priming" action is. Maybe an empty file, then fill it in a second PR?

**Comment by robin-aws** [comment]
I admit I don’t quite see why only existing configured actions are live. Safety doesn’t seem to be a valid argument since malicious actors could just submit PR’s that mess with existing actions instead. Or are there other guard rails I’m not aware of?

**Comment by mattsb42-aws** [comment]
> I admit I don’t quite see why only existing configured actions are live. Safety doesn’t seem to be a valid argument since malicious actors could just submit PR’s that mess with existing actions instead. Or are there other guard rails I’m not aware of?

It's not a perfect mechanism, there might be other reasons, and my guess about why it works that way might be wrong. It's also not really relevant, though. It's not something we can do anything about, it's just how the tool works.

**Comment by mattsb42-aws** [comment]
side note: as I mentioned in #135, which is priming the repo so that these workflows will actually run in this PR, I've removed the branch requirement for Travis to pass because it looks like it's broken at the moment and I'd rather invest in moving to GitHub Actions than fixing Travis.

File: .github/workflows/integ-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Local Tests
```
Comment (dougch) [inline .github/workflows/integ-tests.yaml]: name: Integ Tests

File: .github/workflows/integ-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          - os: ubuntu-latest
+            architecture: x86
+          - os: macos-latest
+            architecture: x86
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+          architecture: ${{ matrix.architecture }}
```
Comment (dougch) [inline .github/workflows/integ-tests.yaml]: Is this implicit?  Shouldn't there be an architecture in the matrix?

File: .github/workflows/local-tests.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          - os: ubuntu-latest
+            architecture: x86
+          - os: macos-latest
+            architecture: x86
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+          architecture: ${{ matrix.architecture }}
```
Comment (dougch) [inline .github/workflows/local-tests.yaml]: same as above, where is this defined?

**Comment by mattsb42-aws** [comment]
this was taken care of with #142

**Pull request closed without merging**